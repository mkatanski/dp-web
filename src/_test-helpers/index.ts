export * from "./withResponsiveContext";
export * from "./withThemeProvider";
