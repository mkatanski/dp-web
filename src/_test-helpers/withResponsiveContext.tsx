import React from "react";
import { Context as ResponsiveContext } from "react-responsive";
import { Breakpoints } from "config/breakpoints";

export const withMobileSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.mobile }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withTabletSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.tablet }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.desktop }}>
    {Component}
  </ResponsiveContext.Provider>
);

export const withLargeDesktopSize = (Component: React.ReactNode) => (
  <ResponsiveContext.Provider value={{ width: Breakpoints.largeDesktop }}>
    {Component}
  </ResponsiveContext.Provider>
);
