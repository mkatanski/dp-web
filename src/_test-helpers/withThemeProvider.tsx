import React from "react";
import { ThemeProvider as DefaultThemeProvider } from "lib/theme/ThemeProvider";
import { StyledTheme, Theme } from "styles/theme/_types";
import { createMuiTheme } from "@material-ui/core";

const testTheme: StyledTheme = {
  primary: "primary",
  secondary: "secondary",
  textPrimary: "textPrimary",
  textSecondary: "textSecondary",
  textInverse: "textInverse"
};

const muiTheme = createMuiTheme({
  direction: "ltr"
});

const theme: Theme = Object.assign(testTheme, muiTheme);

export const withThemeProvider = (Component: React.ReactNode) => (
  <DefaultThemeProvider theme={theme}>{Component}</DefaultThemeProvider>
);
