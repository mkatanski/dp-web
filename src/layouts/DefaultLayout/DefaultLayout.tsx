import React from "react";
import styled from "styled-components";

import { mediaSize } from "styles/theme/mediaQueries";
import { SidebarStateProvider } from "lib/sidebar";

import { Container as BareContainer } from "@material-ui/core";
import LayoutBase, { LayoutBaseProps } from "layouts/LayoutBase";
import { TopBar } from "components/TopBar";
import { Sidebar } from "components/Sidebar";
import { NavListView } from "views/NavListView";
import { usePendingTimer } from "hooks/usePendingTimer";

const Container = styled(BareContainer)`
  min-height: 100vh;
  display: flex;
  padding: 0;

  @media all and (-ms-high-contrast: none) {
    height: 0;
  }
`;

const Content = styled(BareContainer)`
  flex-grow: 1;
  max-width: 100%;
  padding: ${({ theme }) => `${theme.spacing(8, 0, 0, 0)}`};

  ${mediaSize.largeDesktop} {
    padding: ${({ theme }) => `${theme.spacing(8, 0, 0, 32)}`};
  }
`;

const PageContent = styled(BareContainer)`
  padding-top: ${({ theme }) => theme.spacing(3)}px;
  padding-bottom: ${({ theme }) => theme.spacing(3)}px;
  height: 100%;
  display: flex;
  flex-direction: column;
`;

const DefaultLayout: React.FC<LayoutBaseProps> = ({
  children,
  ...rest
}: LayoutBaseProps) => {
  usePendingTimer();
  return (
    <SidebarStateProvider>
      <TopBar>
        <h3>Template Deployments</h3>
      </TopBar>
      <Sidebar>
        <NavListView />
      </Sidebar>
      <Container maxWidth={false}>
        <Content maxWidth={false}>
          <PageContent maxWidth={false}>
            <LayoutBase {...rest}>{children}</LayoutBase>
          </PageContent>
        </Content>
      </Container>
    </SidebarStateProvider>
  );
};

export default DefaultLayout;
