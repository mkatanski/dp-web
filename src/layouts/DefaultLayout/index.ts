import DefaultLayout from "./DefaultLayout";
export * from "./DefaultLayout";

export default DefaultLayout;
