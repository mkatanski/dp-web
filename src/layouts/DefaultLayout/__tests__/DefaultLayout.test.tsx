/* eslint-disable react/display-name */
import React from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";

import { withThemeProvider } from "_test-helpers";
import DefaultLayout from "../DefaultLayout";

jest.mock("react-router-dom", () => ({
  useLocation: () => ({ pathname: "testPath" })
}));

jest.mock(
  "layouts/LayoutBase",
  () => ({ children, ...rest }: React.PropsWithChildren<{}>) => (
    <div id="LayoutBase" {...rest}>
      {children}
    </div>
  )
);

jest.mock("components/TopBar", () => ({
  TopBar: ({ children }: React.PropsWithChildren<{}>) => (
    <div id="TopBar">{children}</div>
  )
}));

jest.mock("components/Sidebar", () => ({
  Sidebar: ({ children, ...rest }: React.PropsWithChildren<{}>) => (
    <div id="Sidebar" data-props={rest}>
      {children}
    </div>
  )
}));

jest.mock("views/NavListView", () => ({
  NavListView: ({ ...rest }) => <div id="NavListView" data-props={rest} />
}));

jest.mock("hooks/usePendingTimer", () => ({
  usePendingTimer: jest.fn()
}));

describe("layouts", () => {
  describe("DefaultLayout", () => {
    it("should render DefaultLayout with children", () => {
      const tree = TestRenderer.create(
        withThemeProvider(<DefaultLayout>Some children</DefaultLayout>)
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
