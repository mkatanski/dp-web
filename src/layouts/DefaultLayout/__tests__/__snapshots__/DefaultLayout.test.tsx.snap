// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layouts DefaultLayout should render DefaultLayout with children 1`] = `
Array [
  @media all and (-ms-high-contrast:none) {

}

@media (min-width:1200px) {

}

<div
    id="TopBar"
  >
    <h3>
      Template Deployments
    </h3>
  </div>,
  @media all and (-ms-high-contrast:none) {

}

@media (min-width:1200px) {

}

<div
    data-props={Object {}}
    id="Sidebar"
  >
    <div
      data-props={Object {}}
      id="NavListView"
    />
  </div>,
  .c0 {
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  max-width: 100%;
  padding: 64px 0px 0px 0px;
}

.c2 {
  padding-top: 24px;
  padding-bottom: 24px;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media all and (-ms-high-contrast:none) {
  .c0 {
    height: 0;
  }
}

@media (min-width:1200px) {
  .c1 {
    padding: 64px 0px 0px 256px;
  }
}

<div
    className="c0"
    data-props={
      Object {
        "maxWidth": false,
      }
    }
    id="MUI.Container"
  >
    <div
      className="c1"
      data-props={
        Object {
          "maxWidth": false,
        }
      }
      id="MUI.Container"
    >
      <div
        className="c2"
        data-props={
          Object {
            "maxWidth": false,
          }
        }
        id="MUI.Container"
      >
        <div
          id="LayoutBase"
        >
          Some children
        </div>
      </div>
    </div>
  </div>,
]
`;
